src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -472,9 +558,17 @@ internal InteractionResult() { }
 
     public partial interface IProjectMetadata : ApplicationModel.IResourceAnnotation
     {
+        bool IsFileBasedApp { get; }
```

Comment:
@DamianEdwards - is this a good name? Should it be `IsSingleFileBasedApp`? Aren't all apps "file based"?